Chp12Lsn7/UnitTestProject/UnitTest.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Chp12Lsn7;
+using UserCommunication;
+
+namespace UnitTestProject
+{
+    [TestClass]
+    public class UnitTest
+    {
+        [TestMethod]
+        [ExpectedException(typeof(System.ArgumentException), "Invalid Input! Cannot enter a number less than or equal to zero.")]
+        public void SqrtTest()
+        {
+            int[] invalidNumbers = { 0, -2 };
+            int numberOfColumns = invalidNumbers.GetLength(0);
+
+            for (int i = 0; i < numberOfColumns; i++)
+            {
+                int testNumbers = invalidNumbers[i];
+                Program.CalculateSquareRoot(testNumbers);
+            }
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(System.ArgumentException), "Invalid Input! Cannot enter a number less than or equal to zero.")]
+        public void UserInputTestArgument()
+        {
+            string[] invalidNumbers = { "0", "-2" };
+            int numberOfColumns = invalidNumbers.GetLength(0);
+
+            for (int i = 0; i < numberOfColumns; i++)
+            {
+                string testNumbers = invalidNumbers[i];
+                UserBarrier.UserInput(testNumbers);
+            }
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(System.FormatException))]
+        public void UserInputTestFormat()
+        {
+            string[] invalidNumbers = { "a", "apple" };
+            int numberOfColumns = invalidNumbers.GetLength(0);
+
+            for (int i = 0; i < numberOfColumns; i++)
+            {
+                string testNumbers = invalidNumbers[i];
+                UserBarrier.UserInput(testNumbers);
+            }
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(System.OverflowException))]
+        public void UserInputTestOverflow()
+        {
+            string[] invalidNumbers = { "2147483648", "–2147483649" };
+            int numberOfColumns = invalidNumbers.GetLength(0);
+
+            for (int i = 0; i < numberOfColumns; i++)
+            {
+                string testNumbers = invalidNumbers[i];
+                UserBarrier.UserInput(testNumbers);
+            }
+        }
+
+        [TestMethod]
+        public void SquareRootCorrectness()
+        {
+            int[] positiveInput = { 4, 16, 256 };
+            int[] correctOutput = { 2, 4, 16 };
+            int inputColumns = positiveInput.GetLength(0);
+            int outputColumns = correctOutput.GetLength(0);
+
+            for (int i = 0; i < inputColumns && i < outputColumns; i++)
+            {
+                int inputNumbers = positiveInput[i];
+                int outputNumbers = correctOutput[i];
+
+                double finalAnswer = Program.CalculateSquareRoot(inputNumbers);
+                Assert.IsTrue(finalAnswer == outputNumbers);
+            }
+        }
+
```

Comment:
Consider:

    public class InOut {
        public string In;
        public int Out;
    }

    //With the above you can then do...

    InOut[] io = new InOut[] { 
                                            new InOut{In="1",Out=1}, 
                                            new InOut{In="2", Out=2}
                        };